### ISSUE TITLE
On Dialog dismissed returns PermissionStatus.permanentlyDenied

### ISSUE THREAD

**Issue by eli1stark** [Baseflow/flutter-permission-handler#757]
## 🐛 Bug Report
I noticed strange behavior. When I dismiss a permission dialog it returns `PermissionStatus.permanentlyDenied`

### Expected behavior
When dissmising a permission dialog it must return `PermissionStatus.denied`.
### Reproduction steps
1. `final permission = await Permission.contacts.request()`
2. Dissmis dialog (tap on the grayed area) and check for permission status.

### Configuration

**Version:** 8.3.0

**Platform:** 
- [ ] :iphone: iOS
- [x] :robot: Android

**Comment by VovaMedvediev** [comment]
+

**Comment by VovaMedvediev** [comment]
the same to me

**Comment by marcelo-bottoni** [comment]
I have the same issue here.

Tested today with versions 9.2.0 and 10.0.0 and still have the same issue.
Faced this problem on Android.

Here are some additional details I observed during my tests:

- If the permission is set as "**ASK EVERY TIME**", or if I have never requested it before, when I dismiss the dialog I get **PermissionStatus.permanentlyDenied**.

- Now if I have already requested and denied the permission one time already (not two), I dismiss the dialog I get **PermissionStatus.denied**.
